bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -224,9 +205,9 @@ private ObjectId(final int timestamp, final int machineIdentifier, final short p
             throw new IllegalArgumentException("The counter must be between 0 and 16777215 (it must fit in three bytes).");
         }
         this.timestamp = timestamp;
-        this.machineIdentifier = machineIdentifier;
-        this.processIdentifier = processIdentifier;
         this.counter = counter & LOW_ORDER_THREE_BYTES;
+        this.randomValue1 = machineIdentifier;
+        this.randomValue2 = (int) processIdentifier;
```

Comment:
Unnecessary cast?

Reply:
`randomValue2` is now a short, so no cast necessary.